ush/python/pygfs/jedi/jedi.py

```diff
@@ -4,145 +4,177 @@
+        self.link_exe()
 
     @logit(logger)
-    def execute(self, task_config: AttrDict, aprun_cmd: str, jedi_args: Optional[List] = None) -> None:
+    def execute(self) -> None:
         """Execute JEDI application
 
         Parameters
         ----------
-        task_config: AttrDict
-            Attribute-dictionary of all configuration variables associated with a GDAS task.
-        aprun_cmd: str
-            String comprising the run command for the JEDI executable.
-        jedi_args (optional): List
-            List of strings comprising optional input arguments for the JEDI executable.
+        None
 
         Returns
         ----------
-        jedi_config: AttrDict
-            Attribute-dictionary of JEDI configuration rendered from a template.
+        None
         """
 
-        chdir(task_config.DATA)
+        chdir(self.jedi_config.rundir)
 
-        exec_cmd = Executable(aprun_cmd)
-        exec_cmd.add_default_arg(self.exe)
-        if jedi_args:
-            for arg in jedi_args:
+        exec_cmd = Executable(self.jedi_config.mpi_cmd)
+        exec_cmd.add_default_arg(self.jedi_config.exe)
+        if self.jedi_config.jedi_args:
+            for arg in self.jedi_config.jedi_args:
                 exec_cmd.add_default_arg(arg)
-        exec_cmd.add_default_arg(self.yaml)
+        exec_cmd.add_default_arg(self.jedi_config.yaml)
 
+        logger.info(f"Executing {exec_cmd}")
         try:
             exec_cmd()
         except OSError:
             raise OSError(f"FATAL ERROR: Failed to execute {exec_cmd}")
         except Exception:
             raise WorkflowException(f"FATAL ERROR: An error occurred during execution of {exec_cmd}")
 
-    @staticmethod
     @logit(logger)
-    def link_exe(task_config: AttrDict) -> None:
+    def render_jcb(self, task_config: AttrDict, algorithm: Optional[str] = None) -> AttrDict:
+        """Compile a JEDI configuration dictionary from a template file and save to a YAML file
+
+        Parameters
+        ----------
+        task_config : AttrDict
+            Dictionary of all configuration variables associated with a GDAS task.
+        algorithm (optional) : str
+            Name of the algorithm used to generate the JEDI configuration dictionary.
+            It will override the algorithm set in the jedi_config.jcb_algo_yaml file.
+
+        Returns
+        ----------
+        jedi_input_config: AttrDict
+            Attribute-dictionary of JEDI configuration rendered from a template.
+        """
+
+        # Fill JCB base YAML template and build JCB config dictionary
+        if self.jedi_config.jcb_base_yaml:
+            jcb_config = parse_j2yaml(self.jedi_config.jcb_base_yaml, task_config)
+        else:
+            raise KeyError(f"FATAL ERROR: JEDI configuration dictionary must contain jcb_base_yaml.")
+
+        # Add JCB algorithm YAML, if it exists, to JCB config dictionary
+        if self.jedi_config.jcb_algo_yaml:
```

Comment:
same as above.

Reply:
@aerorahul Same as my comment above

Reply:
@aerorahul An exception shouldn't be raised here if ```jcb_algo_yaml``` is set to ```None```. JCB algorithm YAMLs are not required. It's only required that the key, ```algorithm```, is present somewhere in the jcb_config dict, and it's only sometimes specified in the JCB algorithm YAML.